packages/hoprd/src/index.ts

```diff
@@ -171,6 +178,28 @@ async function generateNodeOptions(): Promise<HoprOptions> {
     options.dbPath = argv.data
   }
 
+  const protocolConfig = require('../protocol-config.json') as ProtocolConfig
+  for (const environment of protocolConfig.environments) {
+    if (environment.id === argv.environment) {
+      for (const network of protocolConfig.networks) {
+        if (network.id === environment.network_id) {
+          options.environment = {
+            id: environment.id,
+            network,
+            channel_contract_deploy_block: environment.channel_contract_deploy_block,
+            token_contract_address: environment.token_contract_address,
+            channels_contract_address: environment.channels_contract_address
+          }
+        }
+      }
+
+    }
+  }
+
+  if (!options.environment) {
```

Comment:
The search algorithm isn't documented such that a user would understand it. It would also make sense to be able to list the possible values for the environment parameter as part of `--help``.